services/paymaster/src/constraint.rs

```diff
@@ -95,8 +125,10 @@ impl VariationOrderedInstructionConstraints {
                 if constraint.required {
                     return result;
                 }
+                constraint_index += 1;
             } else {
                 instruction_index += 1;
+                constraint_index += 1;
```

Comment:
This `constraint_index += 1;` could be outside of the if else

Reply:
to me, the explicitness makes it more readable. otherwise, it's more mental effort to process what happens to which indices depending on the result and the `required` bool

Reply:
got it